lib/sentry/opentelemetry/span_processor.ex

```diff
@@ -0,0 +1,199 @@
+if Code.ensure_loaded?(OpenTelemetry) do
+  defmodule Sentry.OpenTelemetry.SpanProcessor do
+    @moduledoc false
+
+    require OpenTelemetry.SemConv.ClientAttributes, as: ClientAttributes
+    require OpenTelemetry.SemConv.Incubating.DBAttributes, as: DBAttributes
+    require OpenTelemetry.SemConv.Incubating.HTTPAttributes, as: HTTPAttributes
+    require OpenTelemetry.SemConv.Incubating.URLAttributes, as: URLAttributes
+    require OpenTelemetry.SemConv.Incubating.MessagingAttributes, as: MessagingAttributes
+    @behaviour :otel_span_processor
+
+    require Logger
+
+    alias Sentry.{Transaction, OpenTelemetry.SpanStorage, OpenTelemetry.SpanRecord}
+    alias Sentry.Interfaces.Span
+
+    @impl true
+    def on_start(_ctx, otel_span, _config) do
```

Comment:
The way you've implemented it, we can do nothing in `on_start` since the `on_end` logic is sufficient to build the span tree and transaction.

This will also simplify your `SpanStorage.update_span` where you can remove the unnecessary deletes.
(This is also how we've done it in the new python and JS span processors.)

Reply:
@sl0thentr0py I'll dig into this more but when I make on_start a no-op the tests are no longer passing 🙃

Reply:
@whatyouhide yes but not until now I managed to figure out how to do it to make this work, I just pushed 9a5c550cd77c6b00990f39f74c87a93c6601b740 that simplified things